[4.0] Fix form data lost when create category fail

joomdonation opened the pull request:
Pull Request for Issue #32829.

### Summary of Changes
This PR fix issue https://github.com/joomla/joomla-cms/issues/32829. **category** view uses different key to get form data from session, so we need to override save method in the controller to set this data using the right key.


### Testing Instructions
1. Access to Content -> Categories, create a new category
2. Enter an existing Alias into Alias field (so that category could not be saved)
3. Apply patch
4. Before patch: Error message displayed, but form data lost when you are redirected back to add category screen
5. After patch: Error message displayed, you are redirected back to add category screen but the data you entered before is kept so that you don't have to re-enter.
6. Click on Cancel button to cancel adding the category with that same Alias, you will be redirected back to Categories List screen.
7. Press New button to add new category again, make sure you don't see the old data from previous failed add.

joomdonation committed:
Fix category form data lost

joomdonation committed:
Typo

ceford commented:
I have tested this item :white_check_mark: successfully on d9d8e03e41beb4be19d5556ec7a644dc6a5f48dc<br /><br />Works as described.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32959">issues.joomla.org/tracker/joomla-cms/32959</a>.</sub>

alikon commented:
I have tested this item :white_check_mark: successfully on d9d8e03e41beb4be19d5556ec7a644dc6a5f48dc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32959">issues.joomla.org/tracker/joomla-cms/32959</a>.</sub>

alikon commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32959">issues.joomla.org/tracker/joomla-cms/32959</a>.</sub>

SharkyKZ commented:
I have tested this item :red_circle: unsuccessfully on d9d8e03e41beb4be19d5556ec7a644dc6a5f48dc<br /><br />Data remains after cancelling a failed edit.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32959">issues.joomla.org/tracker/joomla-cms/32959</a>.</sub>

joomdonation committed:
Clear form data from session for failed edit action

joomdonation commented:
Thanks @SharkyKZ for reviewing the PR, didn't think about that cancel action. I updated the code and should fix your reported issue now. I also updated testing instructions to include your test case.

@ceford @alikon When you have time, please help testing it again.

alikon commented:
I have tested this item :white_check_mark: successfully on 9815e11ca1cb0d13ff3aae682684c38e65c0223a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32959">issues.joomla.org/tracker/joomla-cms/32959</a>.</sub>

ceford commented:
I have tested this item :white_check_mark: successfully on 9815e11ca1cb0d13ff3aae682684c38e65c0223a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32959">issues.joomla.org/tracker/joomla-cms/32959</a>.</sub>

alikon commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32959">issues.joomla.org/tracker/joomla-cms/32959</a>.</sub>

richard67 commented:
Thanks!
